src/libtiled/wangset.cpp

```diff
@@ -301,6 +319,132 @@ WangSet::WangSet(Tileset *tileset,
     mCornerColors(cornerColors),
     mUniqueFullWangIdCount(0)
 {
+    Q_ASSERT(mEdgeColors <= 15 && mCornerColors <= 15);
+
+    if (mEdgeColors > 1) {
```

Comment:
There is an empty element at index zero to make indexing a bit cleaner later on, and having just one color means no WangColors should actually be stored. I suppose the condition could be removed, and it would just default to storing the empty color, but is there any value in this?